### PR TITLE
porting enforcerbundle/debug bin from master

### PR DESCRIPTION
**Pull request by amitlimaye**
----

#### Description
*Changes proposed in this pull request.*

#### Test plan
*Outline the test plan used to test this change before merging it.*

> Fixes #.

### CODE REVIEW

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2205",
    "commit-sha": "bc32bcfb591c7469309b18e0921f02e81240e94b",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "1051",
    "commit-sha": "a8ed266cf1cb40466352b5f10aa361aa321e11d0",
    "pipeline": "master"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "1051",
    "commit-sha": "a8ed266cf1cb40466352b5f10aa361aa321e11d0",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2205",
    "commit-sha": "bc32bcfb591c7469309b18e0921f02e81240e94b",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "1051",
    "commit-sha": "a8ed266cf1cb40466352b5f10aa361aa321e11d0",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2205",
    "commit-sha": "8b5b56b7cdc7ca10dd1a56b062a3cc1f32a2ea59",
    "pipeline": "release-3.14.0"
  }
]
```

File: controller/pkg/counters/types.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ const (
 // CounterType custom counter error type.
 type CounterType int
 
+//go:generate stringer -type=CounterType -trimprefix Err
```
Comment (philipatl) [inline controller/pkg/counters/types.go:20]: Is there a `go generate` in a Makefile somewhere? Or do you have to run it manually in the future?

**Review by philipatl — APPROVED** [review]

**Comment by amitlimaye** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "enforcerd",
    "pr-id": "2205",
    "commit-sha": "bd4a95594e3c5f201d1fee72ded89766aab6b002",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "",
    "component": "trireme-lib",
    "pr-id": "1051",
    "commit-sha": "a8ed266cf1cb40466352b5f10aa361aa321e11d0",
    "pipeline": "release-3.14.0"
  }
]
```

File: controller/pkg/counters/types.go (hunk 1 above)
Comment (amitlimaye) [reply controller/pkg/counters/types.go:20]: Run it manually go generate will not walk vendor directory which is where the trireme-lib resides. 
Everytime we modufy counters we need to generate the countertypes_string.go file and check it in

**Review by sibicramesh — APPROVED** [review]

**Pull request merged**